### DISCUSSION TITLE
Integrate with vscode

### DISCUSSION THREAD

**Discussion by harrygr** [postmodern/chruby#480] [Q&A]
I'm having some trouble with a VSCode extension that executes a ruby command: [htmlbeautifier](https://github.com/threedaymonk/htmlbeautifier) (via the [vscode-erb-beautify](https://github.com/aliariff/vscode-erb-beautify) extension).

I have no trouble executing the `htmlbeautifier` command from my shell (having chruby'd to the correct version), but vscode errors with `couldn't run htmlbeautifier 'spawn htmlbeautifier ENOENT'`, presumably because it's still using the system ruby.

Any ideas on how to get this working?

**Comment by postmodern** [comment]
This is likely due to VSCode not executing commands in a shell, but executing them as separate processes (JetBrains RubyMine used to do this).

One workaround appears to be installing the [Ruby](https://code.visualstudio.com/docs/editor/extension-marketplace) VSCode extension by Peng Lv, which supports detecting the Ruby via RVM, rbenv, chruby, and asdf.

Another workaround would be to try executing VSCode from the terminal in your project directory, so that in inherits the `PATH`, `GEM_HOME`, etc, set by `chruby`.

**Reply by harrygr** [reply]
Many thanks for the response. I actually have that extension installed, but it only seems to work sporadically. An exit of vscode and executing in the project dir seems to have done the trick for now. Will see if the continues to work.

**Comment by mfa777** [comment]
**Problem:** Prettier was getting stuck while formatting code in a GUI-opened VSCode, and the root cause was found to be a bash environment issue.

**Solution:**

1.  Create a wrapper script named `vscode-launcher.sh`:

    ```bash
    #!/bin/bash
    source /usr/local/share/chruby/chruby.sh
    chruby ruby-3.3.5
    /usr/share/code/code "$@"
    ```

2.  Update the launcher command to use this script. Replace `/usr/share/code/code %F` with `"path/to/vscode-launcher.sh %F"`.
